### PR TITLE
refactor(internal): switch to container.MergeEnv()

### PR DESCRIPTION
**Pull request by jbrockopp**
Dependent on https://github.com/go-vela/types/pull/151

This updates the way we populate environment variables in the pipeline containers to use `container.MergeEnv()`

The benefits of this function are provided in the above dependent PR.

### CODE REVIEW

File: internal/step/environment.go [hunk 1]
```diff
@@ -45,33 +45,24 @@ func Environment(c *pipeline.Container, b *library.Build, r *library.Repo, s *li
 		// populate environment variables from build library
 		//
 		// https://pkg.go.dev/github.com/go-vela/types/library#Build.Environment
-		c.Environment = appendMap(c.Environment, b.Environment(workspace, channel))
+		c.MergeEnv(b.Environment(workspace, channel))
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply internal/step/environment.go]: fixed 👍

File: internal/step/environment.go [hunk 2]
```diff
@@ -45,33 +45,24 @@ func Environment(c *pipeline.Container, b *library.Build, r *library.Repo, s *li
...
 	}
 
 	// check if the repo provided is empty
 	if r != nil {
 		// populate environment variables from build library
 		//
 		// https://pkg.go.dev/github.com/go-vela/types/library#Repo.Environment
-		c.Environment = appendMap(c.Environment, r.Environment())
+		c.MergeEnv(r.Environment())
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply internal/step/environment.go]: fixed 👍

File: internal/step/environment.go [hunk 3]
```diff
@@ -45,33 +45,24 @@ func Environment(c *pipeline.Container, b *library.Build, r *library.Repo, s *li
...
 	}
 
 	// check if the step provided is empty
 	if s != nil {
 		// populate environment variables from step library
 		//
 		// https://pkg.go.dev/github.com/go-vela/types/library#Service.Environment
-		c.Environment = appendMap(c.Environment, s.Environment())
+		c.MergeEnv(s.Environment())
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply internal/step/environment.go]: fixed 👍

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Review by wass3r — APPROVED** [review]

**Review by wass3r — APPROVED** [review]

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Pull request merged**